clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/LoadBalancerV2UpsertHandler.groovy

```diff
@@ -44,45 +52,72 @@ class LoadBalancerV2UpsertHandler {
+    if(deployDefaults !=null) {
+      if (attributes) {
+        if (TargetTypeEnum.Lambda.toString().equalsIgnoreCase(targetGroup.getTargetType()))
+        {
+          def multiValueHeaderAttribute = attributes.multiValueHeadersEnabled ?: MULTI_VALUE_HEADERS_ENABLED
+          targetGroupAttributes.add(new TargetGroupAttribute(key: "lambda.multi_value_headers.enabled", value: multiValueHeaderAttribute))
+
+        } else {
+          Integer deregistrationDelay = [attributes.deregistrationDelay, deployDefaults?.loadBalancing?.deregistrationDelayDefault].findResult(Closure.IDENTITY)
+
+          def deregistrationDealyAttribute = deregistrationDelay?.toString() ?: DE_REGISTRATION_DELAY
+          targetGroupAttributes.add(new TargetGroupAttribute(key: "deregistration_delay.timeout_seconds", value: deregistrationDealyAttribute))
           }
-          if (attributes.stickinessType != null) {
-            targetGroupAttributes.add(new TargetGroupAttribute(key: "stickiness.type", value: attributes.stickinessType))
+          if (loadBalancer.type == 'application') {
+            def stickinessEnabledAttribute = attributes.stickinessEnabled?.toString() ?: STICKINESS_ENABLED
+            targetGroupAttributes.add(new TargetGroupAttribute(key: "stickiness.enabled", value: stickinessEnabledAttribute))
+
+            def stickinessTypeAttribute = attributes.stickinessType ?: STICKINESS_TYPE
+              targetGroupAttributes.add(new TargetGroupAttribute(key: "stickiness.type", value: stickinessTypeAttribute))
+
+            def stickinessDurationAttribute = attributes.stickinessDuration?.toString() ?: STICKINESS_DURATION
+            targetGroupAttributes.add(new TargetGroupAttribute(key: "stickiness.lb_cookie.duration_seconds", value: stickinessDurationAttribute))
+
           }
-          if (attributes.stickinessDuration != null) {
-            targetGroupAttributes.add(new TargetGroupAttribute(key: "stickiness.lb_cookie.duration_seconds", value: attributes.stickinessDuration.toString()))
+          if (loadBalancer.type == 'network' ) {
+            def proxyProtocolV2Attribute = attributes.proxyProtocolV2 ?: PROXY_PROTOCOL_V2
+            targetGroupAttributes.add(new TargetGroupAttribute(key: "proxy_protocol_v2.enabled", value: proxyProtocolV2Attribute))
+
+            }
           }
         }
-        if (loadBalancer.type == 'network' ) {
-          if(attributes.proxyProtocolV2 != null) {
-            boolean currentProxyProtocolV2 = currentTargetGroupAttributes.find { it.key == ATTRIBUTE_PROXY_PROTOCOL_V2_ENABLED }?.getValue().toBoolean()
-            if (currentProxyProtocolV2 != null && currentProxyProtocolV2 != attributes.proxyProtocolV2) {
-              targetGroupAttributes.add(new TargetGroupAttribute(key: "proxy_protocol_v2.enabled", value: currentProxyProtocolV2))
+    else {
+      // Modify Target Group Attributes
+      if (attributes) {
```

Comment:
attributes may be not-null with all the values null, you might need to track that you've actually added something to targetGroupAttributes before issuing the command to update

Reply:
sounds good , will fix that